classes/privacy/provider.php

```diff
@@ -25,19 +25,240 @@
+        $collection->add_database_table(
+                'auth_enrolkey_redirect',
+                [
+                        'usermodified' => 'privacy:metadata:auth_enrolkey_redirect:usermodified',
+                ],
+                'privacy:metadata:auth_enrolkey_redirect'
+        );
+        $collection->add_database_table(
+                'auth_enrolkey_profile',
+                [
+                        'usermodified' => 'privacy:metadata:auth_enrolkey_profile:usermodified',
+                ],
+                'privacy:metadata:auth_enrolkey_profile'
+        );
+        $collection->add_database_table(
+                'auth_enrolkey_cohort',
+                [
+                        'usermodified' => 'privacy:metadata:auth_enrolkey_cohort:usermodified',
+                ],
+                'privacy:metadata:auth_enrolkey_cohort'
+        );
+
+        return $collection;
     }
-}
\ No newline at end of file
+
+    /**
+     * Get the list of contexts that contain user information for the given user.
+     *
+     * @param int $userid the userid to search.
+     * @return contextlist the contexts in which data is contained.
+     */
+    public static function get_contexts_for_userid(int $userid): contextlist {
+        $sql = "SELECT ctx.id
+                  FROM {user_enrolments} ue
+                  JOIN {enrol} e
+                    ON e.id = ue.enrolid
+                   AND ue.userid = :userid
+                  JOIN {context} ctx
+                    ON ctx.instanceid = e.courseid
+                   AND ctx.contextlevel = :contextlevel
+                  JOIN {auth_enrolkey_redirect} aer
+                    ON aer.usermodified = ue.userid
+                  JOIN {auth_enrolkey_profile} aep
+                    ON aep.usermodified = ue.userid
+                  JOIN {auth_enrolkey_cohort} aec
+                    ON aec.usermodified = ue.userid";
+        $params = [
+                'contextlevel' => CONTEXT_USER,
+                'userid'       => $userid
+        ];
+        $contextlist = new contextlist();
+        $contextlist->add_from_sql($sql, $params);
+
+        return $contextlist;
+    }
+
+    /**
+     * Gets the list of users who have data with a context.
+     *
+     * @param userlist $userlist the userlist containing users who have data in this context.
+     */
+    public static function get_users_in_context(userlist $userlist) {
+        $context = $userlist->get_context();
+
+        if (!$context instanceof \context_user) {
+            return;
+        }
+
+        // If current context is course, all users are contained within, get all users.
+        if ($context->contextlevel == CONTEXT_USER) {
+            $params = [
+                    'contextlevel' => CONTEXT_USER,
+                    'contextid' => $context->id,
+            ];
+
+            $sql = "SELECT usermodified AS userid FROM {auth_enrolkey_redirect} aer
+                        JOIN {enrol} e
```

Comment:
The enrol table doesn't have anything to do with this plugin. This SQL needs to be looked at.